runtime/compiler/CMakeLists.txt

```diff
@@ -27,6 +27,16 @@
 #
 # Longer term, this will of course have to collapse into the VM builds.
 
+if(OMR_ARCH_X86)
+	enable_language(ASM_NASM)
+	# We have to manually set thses since older cmake versions dont handle include directories for NASM properly
```

Comment:
@dnakamura , fyi..

Reply:
@0xdaryl It is worth noting that this little hack to the paths on will not be needed on NASM v 2.14 onwards due to [this patch](https://bugzilla.nasm.us/attachment.cgi?id=411681&action=diff) that went into NASM v 2.14 and lets `nasm_catfile(path, filename)` handle unifying the filepath to the filenames. That's how it looks like:

```c
char *nasm_catfile(const char *dir, const char *file)
{
#ifndef catsep
    return nasm_strcat(dir, file);
#else
    size_t dl = strlen(dir);
    size_t fl = strlen(file);
    char *p, *pp;
    bool dosep = true;

    if (!dl || ismatch(separators, dir[dl-1])) {
        /* No separator necessary */
        dosep = false;
    }

    p = pp = nasm_malloc(dl + fl + dosep + 1);

    memcpy(pp, dir, dl);
    pp += dl;
    if (dosep)
        *pp++ = catsep;

    memcpy(pp, file, fl+1);

    return p;
#endif
}
```

We would then no longer have to worry about the trailing `/`. In the future, once NASM v2.14 can be set as the minimum required, we would be able to clean this part up. I think that could be mentioned in the comments in this file so that the include paths for NASM could be treated the same way as the rest of the project.

Reply:
@dnakamura It seems to me like even the newest versions of CMake would need trailing `/` in the `-I` paths for NASM versions < 2.14 based on the discussions here: https://gitlab.kitware.com/cmake/cmake/issues/12919

Seems like the people at CMake weren't willing to provide a workaround to a problem that NASM should fix, so someone was frustrated enough with CMake+NASM to go fix the issue in NASM him/herself: https://bugzilla.nasm.us/show_bug.cgi?id=3392205#c1 . So the actual fix for the include paths is dependent on the version of NASM used, rather than the version of CMake. If I am not wrong, I guess we can remove this comment and instead mention that we need to append the underscore to ensure compatibility with NASM versions older than 2.14.

Reply:
Yes, I was confused. I thought the issue was resolved in newer cmake versions. But it appears that it is not. Updating the comments to reflect that